### ISSUE TITLE
There is no keybinding to toggle vim/evil mode

### ISSUE THREAD

**Issue by singpolyma** [syl20bnr/spacemacs#13511]
#### Description :octocat:

There is no keybinding to toggle vim/evil mode

#### Reproduction guide :beetle:
- Start Emacs
- SPC t E

*Observed behaviour:* :eyes: :broken_heart:

emacs mode and hybrid mode are options (even if I select hybrid mode it stays like this)

*Expected behaviour:* :heart: :smile:

vim / evil should be an option

#### System Info :computer:
- OS: gnu/linux
- Emacs: 26.1
- Spacemacs: 0.200.13
- Spacemacs branch: master (rev. acf5ce87f)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(haskell emacs-lisp markdown helm git org
         (shell :variables shell-default-position 'full)
         version-control)
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG IMAGEMAGICK SOUND GPM DBUS GSETTINGS NOTIFY ACL LIBSELINUX GNUTLS LIBXML2 FREETYPE M17N_FLT LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 THREADS LIBSYSTEMD LCMS2

**Comment by robbyoconnor** [comment]
@duianto is this related to #13461

**Comment by duianto** [comment]
@robbyoconnor This is unrelated.

This is a UI issue that I also noticed it when I started using Spacemacs, before Treemacs was part of Spacemacs.

The `SPC t E` key binding prefix, always lists these two key bindings:
- e -> emacs (holy-mode)
- h -> hybrid (hybrid-mode)

but they work as dis/enable toggles.

Ideally they should update to show the two options that one isn't using.

---

### There's one or two related issues on the `develop` branch.

When enabling: `SPC t E e` holy-mode
- On the `master` branch, the cursor switches to holy-mode right away as expected.
- On `develop`, the cursor remains in evil-mode and only switches to holy-mode
after one tries to enter insert state (ex: `a` or `i`)

This might be related to why the cursor doesn't change right away.

There's `transient-get-suffix` messages when switching to/from the holy-mode (it doesn't happen for hybrid-mode):
>transient-get-suffix: ' not found in magit-dispatch

and:
>transient-get-suffix: X not found in magit-branch

**Comment by TymonKilich** [comment]
What about `C-z`? It "runs the command evil-emacs-state (found in evil-motion-state-map), which is an interactive compiled Lisp function in ‘evil-states.el’."

Not sure about master but it works in `/develop`.

**Comment by jebotz** [comment]
@duianto wrote:
>     * On `develop`, the cursor remains in evil-mode and only switches to holy-mode
>       after one tries to enter insert state (ex: `a` or `i`) [...]
> There's `transient-get-suffix` messages when switching to/from the holy-mode (it doesn't happen for hybrid-mode): 
> > transient-get-suffix: ' not found in magit-dispatch

This bug is still present; the holy-mode toggle works fine if the git layer is not installed, but if it is I see the described bug and message even in a fresh install of spacemacs (develop branch) with only the git layer added.

**Comment by duianto** [comment]
It seems to be caused by: `evil-magit-revert-popups`
in `evil-magit.el`
https://github.com/emacs-evil/evil-magit/blob/04a4580c6eadc0e2b821a3525687e74aefc30e84/evil-magit.el#L604

This is the how it's called:
`SPC t E e` calls `(spacemacs/toggle-holy-mode)`

The toggle function is defined here:
https://github.com/syl20bnr/spacemacs/blob/64900229a1e020d0b0b942efe112ca968e24aa05/layers/%2Bdistributions/spacemacs-bootstrap/packages.el#L540

When `holy-mode` is `non-nil`, then it calls:
https://github.com/syl20bnr/spacemacs/blob/64900229a1e020d0b0b942efe112ca968e24aa05/layers/%2Bdistributions/spacemacs-bootstrap/local/holy-mode/holy-mode.el#L58

and it calls:
https://github.com/syl20bnr/spacemacs/blob/64900229a1e020d0b0b942efe112ca968e24aa05/layers/%2Bdistributions/spacemacs-bootstrap/local/holy-mode/holy-mode.el#L72-L73

```
spacemacs-editing-style-hook’s value is
(spacemacs//helm-hjkl-navigation spacemacs//company-active-navigation spacemacs//evil-escape-deactivate-in-holy-mode spacemacs//magit-evil-magit-bindings spacemacs/set-evil-search-module)
```
specifically: `spacemacs//magit-evil-magit-bindings`

When the buffer isn't evilified:
https://github.com/syl20bnr/spacemacs/blob/64900229a1e020d0b0b942efe112ca968e24aa05/layers/%2Bsource-control/git/funcs.el#L71-L78

then it calls: `(evil-magit-revert)`

And finally, it calls: `(evil-magit-revert-popups)`
https://github.com/emacs-evil/evil-magit/blob/04a4580c6eadc0e2b821a3525687e74aefc30e84/evil-magit.el#L612

which has the following expression:
```elisp
(with-eval-after-load 'forge
      (transient-suffix-put 'magit-dispatch "@" :key "'"))
```

However, the https://github.com/emacs-evil/evil-magit repository says:
>NOTE [2020-11-24]: This repository is no longer maintained. This package is now a part of evil-collection.

It links to: https://github.com/emacs-evil/evil-collection

It defines the same function `evil-collection-magit-revert-popups` here:
https://github.com/emacs-evil/evil-collection/blob/e7be9f53a05c81ef9e89c2f053ba62048e6afd69/modes/magit/evil-collection-magit.el#L581

The only change in this function, seems to be that the word: `collection` has been added between `evil-` and `-magit-...` in all the `evil-magit-...` commands.

**Comment by duianto** [comment]
A PR is pending with a possible fix:
Improve editing-style toggles #14479

**Comment by duianto** [comment]
I'm unable to reproduce the other issue:
>transient-get-suffix: ' not found in magit-dispatch

Possibly because `evil-magit` has been replaced with the `evil-collection`.
[spacemacs-evil] add evil-collection #14200

**Comment by jebotz** [comment]
> I'm unable to reproduce the other issue: 
> > transient-get-suffix: ' not found in magit-dispatch
> Possibly because `evil-magit` has been replaced with the `evil-collection`.

Yeah, I noticed that this stopped happening right around that time...

**Comment by duianto** [comment]
The following PRs changes have been applied to the `develop` branch:
Improve editing-style toggles #14479

@singpolyma Your initial report was from the Spacemacs `master` branch.
If you've already switched to the `develop` branch (it's currently the default and recommended branch).
Could you test that it works as expected. If it does, then you can close this.
